### PR TITLE
Create hero section

### LINKED ISSUE #8: Home page ( Hero section)

### PR DESCRIPTION
**Pull request by MAlngaawy**
Task #8 

Create the homepage hero section

### CODE REVIEW

File: tailwind.config.js [hunk 1]
```diff
@@ -10,7 +10,15 @@ const {
...
   theme: {
+    extend: {
+      width: {
+        "198": "198px",
+        "696": "696px",
+      },
+      spacing: oneLayerWithPixelAdding('spacing'),
+    },
     colors: {
+      default: "#fafafa",
```
Comment (elhaw) [inline tailwind.config.js]: I think the class for that color is already generated `bg-background`

File: src/modules/HomePage/components/HeroSection/Navbar.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  return (
+    <li className='mx-med cursor-pointer'>
+      <Link href={href}>
+        <Typography align='center' variant='body1'>
+          {text}
+        </Typography>
+      </Link>
+    </li>
+  );
+};
```
Comment (elhaw) [inline src/modules/HomePage/components/HeroSection/Navbar.tsx]: It is good job to create a reusable NavItem component but let us organize the code in a separate folder

File: src/modules/HomePage/components/HeroSection/Navbar.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+          className='p-0'
+          onClick={() => {
+            setShow(true);
+          }}
+        >
+          <Menu />
+        </IconButton>{' '}
+      </span>
+      <div
+        className={classNames(
```
Comment (atwfk) [inline src/modules/HomePage/components/HeroSection/Navbar.tsx]: Use the `clsxm` utility function exported from lib folder,
it uses [clsx](https://www.npmjs.com/package/clsx) which is faster and smaller than [classnames](https://www.npmjs.com/package/classnames)

File: src/modules/HomePage/components/HeroSection/Navbar.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+// create a nav item component with two props
+interface NavProps {
+  href: string;
+  text: string;
+}
+const NavItem = ({ href, text }: NavProps): ReactElement => {
+  return (
+    <li className='mx-med cursor-pointer mobile:my-3'>
+      <Link href={href}>
```
Comment (atwfk) [inline src/modules/HomePage/components/HeroSection/Navbar.tsx:71]: Use the `Link` component exported from `m3ntorship-ui`

File: src/modules/HomePage/components/HeroSection/Navbar.tsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+import type { ReactElement } from 'react';
+import React, { useState } from 'react';
+
+import Logo from './assets/logo.svg';
+import Menu from './assets/menu.svg';
+import X from './assets/x.svg';
+
+export const Navbar = (): ReactElement => {
+  const [show, setShow] = useState(false);
+  return (
```
Comment (atwfk) [inline src/modules/HomePage/components/HeroSection/Navbar.tsx:14]: It's better to move the classes you used here in the jsx return to css file to be more readable

File: src/modules/HomePage/components/HeroSection/index.tsx [hunk 6]
```diff
@@ -1,6 +1,64 @@
...
+// The Size Types
+interface Size {
+  width: number | undefined;
+  height: number | undefined;
+}
 
 export default function HeroSection(): ReactElement {
-  return <Typography variant='h6'>Hero Section</Typography>;
+  const size: Size = useWindowSize();
+  return (
```
Comment (atwfk) [inline src/modules/HomePage/components/HeroSection/index.tsx:14]: It's better to move the classes you used here in the jsx return to css file to be more readable

File: src/modules/HomePage/components/HeroSection/sizeHook.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from 'react';
+
+// Define general type for useWindowSize hook, which includes width and height
+interface Size {
+  width: number | undefined;
+  height: number | undefined;
+}
+
+// Hook
+function useWindowSize(): Size {
```
Comment (atwfk) [inline src/modules/HomePage/components/HeroSection/sizeHook.tsx]: I think it's better to return a flag from this hook, which will be true or false based on the max and min width we will pass as an argument to this hook, something like:
```typescript
const matched = useMedia('(max-width: 900px)'); 
```
take a look at [matchMedia api](https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia)

File: src/modules/HomePage/components/HeroSection/index.tsx [hunk 8]
```diff
@@ -1,6 +1,64 @@
...
+    <div className='hero bg-background'>
+      <div className='relative mx-4 medium:mx-6 medium:px-4 large:mx-8 large:px-0 xxlarge:mx-28'>
+        <span className='absolute z-0 h-full w-1/3  rounded-br-lrg rounded-bl-lrg bg-white shadow-shd-1'></span>
+        <span className='absolute right-0 z-0 h-full w-5/12 rounded-br-lrg rounded-bl-lrg bg-white shadow-shd-1'></span>
```
Comment (atwfk) [inline src/modules/HomePage/components/HeroSection/index.tsx]: the two boxes positioned behind Hero section here don't match Figma UI, ensure the dimensions are the same as in Figma

**Review by atwfk — CHANGES_REQUESTED** [review]
Great efforts :muscle::muscle:
we will align with @elhaw to restructure the project folders and I will update you here or in the meeting if possible.

**Review by atwfk — APPROVED** [review]
LGTM

**Pull request merged**